### PR TITLE
Add code analysis

### PR DESCRIPTION
**Pull request by lyndaidaii**
* Add static code analysis 
   1. Add package reference 
   2. Add global editor config

Address: https://github.com/NuGet/Engineering/issues/3417

### CODE REVIEW

**Review by joelverhagen — APPROVED** [review]

**Review by loic-sharma — APPROVED** [review]

**Comment by lyndaidaii** [comment]
CI build and unit test with VS2019: https://nuget.visualstudio.com/NuGetBuild/_build/results?buildId=74523&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=c6eba180-a01f-47c0-aba7-abca1c90cd3f

File: .nuget/packages.config [hunk 1]
```diff
@@ -2,4 +2,32 @@
...
   <package id="xunit.runner.console" version="2.1.0" />
   <package id="MicroBuild.Core" version="0.3.0" />
+  <package id="Castle.Core" version="3.3.3" targetFramework="net46" />
+  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net46" />
+  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net46" />
+  <package id="Microsoft.AspNet.WebApi.OData" version="5.7.0" targetFramework="net46" />
+  <package id="Microsoft.Data.Edm" version="5.8.4" targetFramework="net46" />
+  <package id="Microsoft.Data.OData" version="5.8.4" targetFramework="net46" />
+  <package id="Microsoft.Web.Xdt" version="2.1.1" targetFramework="net46" />
+  <package id="Moq" version="4.5.21" targetFramework="net46" />
```
Comment (joelverhagen) [inline .nuget/packages.config:12]: Why are these here? This packages.config is used to pull down command-line tools or build stuff and should not have project-specific packages.

Comment (lyndaidaii) [reply .nuget/packages.config:12]: Somehow, project packages.config has trouble in  installing packages. As workaround, I think install packages here might resolve this problem.

Comment (joelverhagen) [reply .nuget/packages.config:12]: Are there alternative workaround? I'm not sure if I understand the problem and solution. Maintaining this list along with the packages.config in each project seems like a headache.
